lib/diagnostic.rb

```diff
@@ -13,28 +13,32 @@ def longest(enum)
   # enumerable supplied (i.e. apply the `+` operator)
   # Assume each entry implements a `+` method
   def sum(enum)
-    # Your code goes here
+    enum.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the numbers enumerable are odd
   def some_odd(numbers)
-    # Your code goes here
+    x = false
+    numbers.each {|num| num.odd? ? x = true : x }
+    return x
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # numbers enumerable are even
   def every_even(numbers)
-    # Your code goes here
+    x=true
+    numbers.each {|num| num.even? ? x : x = false }
+    return x
```

Comment:
As above, this will work; however (since this is, after all, a diagnostic about Enumerables) a better approach might have been `numbers.all? { |num| num.even? }`

This can also be written `numbers.all?(&:even?)` / `numbers.all?(:even?)`